// react-native-paper-dropdown/lib/commonjs/dropdown-item.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = require("react");
var _reactNativePaper = require("react-native-paper");
var _jsxRuntime = require("react/jsx-runtime");
function DropdownItem(props) {
  const {
    option,
    width,
    value,
    onSelect,
    toggleMenu,
    isLast,
    menuItemTestID
  } = props;
  const style = (0, _react.useMemo)(() => ({
    minWidth: width
  }), [width]);
  const theme = (0, _reactNativePaper.useTheme)();
  const titleStyle = (0, _react.useMemo)(() => ({
    color: value === option.value ? theme.colors.primary : theme.colors.onBackground,
    width: width
  }), [option.value, theme.colors.onBackground, theme.colors.primary, value, width]);
  const onPress = () => {
    if (option.value) {
      onSelect?.(option.value);
    }
    toggleMenu();
  };
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)(_jsxRuntime.Fragment, {
    children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(_reactNativePaper.Menu.Item, {
      style: style,
      title: option.label,
      titleStyle: titleStyle,
      contentStyle: style,
      onPress: onPress,
      testID: menuItemTestID
    }), !isLast && /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactNativePaper.Divider, {})]
  });
}
var _default = exports.default = DropdownItem;
