// react-native-paper-dropdown/lib/commonjs/use-dropdown.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = require("react");
var _reactNative = require("react-native");
function useDropdown(maxMenuHeight, maxHeightFraction = 2.5) {
  const [enable, setEnable] = (0, _react.useState)(false);
  const {
    height
  } = (0, _reactNative.useWindowDimensions)();
  const finalMenuHeight = maxMenuHeight ?? height / maxHeightFraction;
  const [dropdownLayout, setDropdownLayout] = (0, _react.useState)({
    x: 0,
    y: 0,
    height: 0,
    width: 0
  });
  const toggleMenu = (0, _react.useCallback)(() => {
    _reactNative.Keyboard.dismiss();
    setEnable(prev => !prev);
  }, []);
  const onLayout = (0, _react.useCallback)(({
    nativeEvent: {
      layout
    }
  }) => {
    setDropdownLayout(layout);
  }, []);
  const menuStyle = (0, _react.useMemo)(() => ({
    width: dropdownLayout.width
  }), [dropdownLayout.width]);
  const scrollViewStyle = (0, _react.useMemo)(() => ({
    maxHeight: finalMenuHeight
  }), [finalMenuHeight]);
  return {
    enable,
    setEnable,
    toggleMenu,
    onLayout,
    menuStyle,
    scrollViewStyle,
    dropdownLayout
  };
}
var _default = exports.default = useDropdown;
