// react-native-paper-dropdown/lib/module/dropdown-item.js
import { useMemo } from 'react';
import { useTheme, Menu, Divider } from 'react-native-paper';
import { Fragment, jsx as _jsx, jsxs as _jsxs } from 'react/jsx-runtime';
function DropdownItem(props) {
  const {
    option,
    width,
    value,
    onSelect,
    toggleMenu,
    isLast,
    menuItemTestID
  } = props;
  const style = useMemo(() => ({
    minWidth: width
  }), [width]);
  const theme = useTheme();
  const titleStyle = useMemo(() => ({
    color: value === option.value ? theme.colors.primary : theme.colors.onBackground,
    width: width
  }), [option.value, theme.colors.onBackground, theme.colors.primary, value, width]);
  const onPress = () => {
    if (option.value) {
      onSelect?.(option.value);
    }
    toggleMenu();
  };
  return /*#__PURE__*/_jsxs(Fragment, {
    children: [/*#__PURE__*/_jsx(Menu.Item, {
      style: style,
      title: option.label,
      titleStyle: titleStyle,
      contentStyle: style,
      onPress: onPress,
      testID: menuItemTestID
    }), !isLast && /*#__PURE__*/_jsx(Divider, {})]
  });
}
export default DropdownItem;
