// react-native-paper-dropdown/lib/module/use-dropdown.js
import { useCallback, useState, useMemo } from 'react';
import { Keyboard, useWindowDimensions } from 'react-native';
function useDropdown(maxMenuHeight, maxHeightFraction = 2.5) {
  const [enable, setEnable] = useState(false);
  const {
    height
  } = useWindowDimensions();
  const finalMenuHeight = maxMenuHeight ?? height / maxHeightFraction;
  const [dropdownLayout, setDropdownLayout] = useState({
    x: 0,
    y: 0,
    height: 0,
    width: 0
  });
  const toggleMenu = useCallback(() => {
    Keyboard.dismiss();
    setEnable(prev => !prev);
  }, []);
  const onLayout = useCallback(({
    nativeEvent: {
      layout
    }
  }) => {
    setDropdownLayout(layout);
  }, []);
  const menuStyle = useMemo(() => ({
    width: dropdownLayout.width
  }), [dropdownLayout.width]);
  const scrollViewStyle = useMemo(() => ({
    maxHeight: finalMenuHeight
  }), [finalMenuHeight]);
  return {
    enable,
    setEnable,
    toggleMenu,
    onLayout,
    menuStyle,
    scrollViewStyle,
    dropdownLayout
  };
}
export default useDropdown;
