// react-native-paper-dates/lib/commonjs/Date/DayNames.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.dayNamesHeight = void 0;
var _reactNative = require("react-native");
var _DayName = _interopRequireDefault(require("./DayName"));
var _reactNativePaper = require("react-native-paper");
var _dateUtils = require("./dateUtils");
var _react = _interopRequireWildcard(require("react"));
var _jsxRuntime = require("react/jsx-runtime");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const dayNamesHeight = exports.dayNamesHeight = 44;
function DayNames({
  disableWeekDays,
  locale,
  startWeekOnMonday
}) {
  const theme = (0, _reactNativePaper.useTheme)();
  const shortDayNames = (0, _react.useMemo)(() => {
    // TODO: wait for a better Intl api ;-)
    const weekdays = [new Date(2020, 7, 2), new Date(2020, 7, 3), new Date(2020, 7, 4), new Date(2020, 7, 5), new Date(2020, 7, 6), new Date(2020, 7, 7), new Date(2020, 7, 8)];
    if (startWeekOnMonday) {
      weekdays.push(weekdays.shift());
    }
    const formatter = new Intl.DateTimeFormat(locale, {
      weekday: 'narrow'
    });
    return weekdays.map(date => formatter.format(date));
  }, [locale, startWeekOnMonday]);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactNative.View, {
    style: [styles.dayNames, {
      backgroundColor: theme.colors.surface
    }],
    pointerEvents: "none",
    children: shortDayNames.filter((_, dayIndex) => (0, _dateUtils.showWeekDay)(dayIndex, disableWeekDays)).map((dayName, i) => /*#__PURE__*/(0, _jsxRuntime.jsx)(_DayName.default, {
      label: dayName
    }, `${dayName}_${i}`))
  });
}
const styles = _reactNative.StyleSheet.create({
  dayNames: {
    alignItems: 'center',
    backgroundColor: '#fff',
    flexDirection: 'row',
    height: dayNamesHeight
  }
});
var _default = exports.default = /*#__PURE__*/(0, _react.memo)(DayNames);
